istioctl/pkg/injector/injector-list.go

```diff
@@ -102,9 +104,16 @@ func injectorListCommand(ctx cli.Context) *cobra.Command {
 				}
 			}
 
-			hooksList, err := client.Kube().AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.Background(), metav1.ListOptions{
-				LabelSelector: "app=sidecar-injector",
-			})
+			var hooksList *admitv1.MutatingWebhookConfigurationList
+			if revisionSpecified != "" {
+				hooksList, err = client.Kube().AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.Background(), metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("app=sidecar-injector, istio.io/rev=%s", revisionSpecified),
+				})
+			} else {
```

Comment:
You can append the labelselector string instead of duplicating the client list request to make the code more clear.

Reply:
My question is: in other commands, the revision flag is used to select the control plane, and we treat an empty flag `""` as the default revision that is related to a specific revision, like in installation. How will you handle the inconsistency between the flags if you want to introduce the revision for resource filtering?

The command was designed to list multiple control planes and could you also provide sample output before and after the changes, so we can see if the results are improved with revision filtering?

Reply:
> You can append the labelselector string instead of duplicating the client list request to make the code more clear.

Modified, thx

Reply:
It’s not about the default value for the flag. In these commands, they actually refer to the `default` revision, but in your case, it refers to `all` revisions

Reply:
I understand. At present, the `--revision` flag in other places is used to specify the control platform, while this one is used to filter webhooks, so the function here is more similar to the `--namespace` flag.
According to the original intention of the command, it lists all webhooks and their targets, so I don't think it needs to be consistent with other places.

Reply:
From your output, it looks like injectors are generating duplicated results, and each of them has all the un-injected namespaces. I think there's a bug related to the namespace matching, which results in duplicated namespaces. Can you try to fix that first? This command might be useful if the results are correct.

Reply:
It seems that because mutatingwebhookconfigurations contains an object injector webhook and a namespace injector webhook, the selector results of both webhooks are printed. The original intention of this command should be to print only the namespaces matched by the namespace injector webhook.

```yaml
apiVersion: admissionregistration.k8s.io/v1
kind: MutatingWebhookConfiguration
...
webhooks:
- admissionReviewVersions:
  ...
  name: rev.namespace.sidecar-injector.istio.io
  namespaceSelector:
...
- admissionReviewVersions:
  ...
  name: rev.object.sidecar-injector.istio.io
  namespaceSelector:
...
```